src/main/scala/org/apache/spark/sql/pulsar/PulsarSourceRDD.scala

```diff
@@ -96,18 +104,20 @@ private[pulsar] abstract class PulsarSourceRDDBase(
                 s"Potential Data Loss: intended to start at $startOffset, " +
                   s"actually we get $currentId")
             }
+            if (isTraceEnabled()) {
+              logTrace(s"First message read has id ${currentId}")
+            }
 
             (startOffset, currentId) match {
               case (_: BatchMessageIdImpl, _: BatchMessageIdImpl) =>
               // we seek using a batch message id, we can read next directly in `getNext()`
               case (_: MessageIdImpl, cbmid: BatchMessageIdImpl) =>
-                // we seek using a message id, this is supposed to be read by previous task since
-                // it's inclusive for the last batch (start, end], so we skip this batch
-                val newStart = new MessageIdImpl(
-                  cbmid.getLedgerId,
-                  cbmid.getEntryId + 1,
-                  cbmid.getPartitionIndex)
-                reader.seek(newStart)
+                // We can only really get to this scenario
+                // when producers sent a batched message will a single record.
+                // The reader will read a batched message but the message id
+                // returned by getLastMessageId() will return a MessageIdImpl.
+                assert(cbmid.getBatchIndex == 0,
```

Comment:
We don't need to seek to the next message as calling readNext on the reader should automatically do that.

Reply:
@poorbarcode we already skip the first here:

https://github.com/streamnative/pulsar-spark/pull/184/files#diff-8b84898f836d0bda3d0943371aae4c1dc40b472ea5aed69cad59401fc8331750R94

Reply:
Sure